src/Symfony/Component/Debug/ExceptionHandler.php

```diff
@@ -189,6 +190,160 @@ public function sendPhpResponse($exception)
+        }
+
+        $content = [
+            'title' => $title,
+            'status' => $statusCode,
+            'detail' => $this->escapeHtml($exception->getMessage()),
+        ];
+
+        if ($this->debug) {
+            $content['exceptions'] = $exception->toArray();
+        }
+
+        return (string) json_encode($content);
+    }
+
+    /**
+     * Gets the XML content associated with the given exception.
+     *
+     * @param \Exception|FlattenException $exception An \Exception or FlattenException instance
+     *
+     * @return string The XML content as a string
+     */
+    public function getXml($exception): string
+    {
+        if (!$exception instanceof FlattenException) {
+            $exception = FlattenException::create($exception);
+        }
+
+        if (404 === $statusCode = $exception->getStatusCode()) {
+            $title = 'Not Found';
+        } elseif (class_exists(Response::class) && isset(Response::$statusTexts[$statusCode])) {
+            $title = Response::$statusTexts[$statusCode];
+        } else {
+            $title = 'Internal Server Error';
+        }
+        $message = $this->escapeHtml($exception->getMessage());
+
+        $exceptions = '';
+        if ($this->debug) {
+            $exceptions .= '<exceptions>';
+            foreach ($exception->toArray() as $e) {
+                $exceptions .= sprintf('<exception class="%s" message="%s"><traces>', $e['class'], $this->escapeHtml($e['message']));
+                foreach ($e['trace'] as $trace) {
+                    $exceptions .= '<trace>';
+                    if ($trace['function']) {
+                        $exceptions .= sprintf('at %s%s%s(%s) ', $trace['class'], $trace['type'], $trace['function'], strip_tags($this->formatArgs($trace['args'])));
+                    }
+                    if (isset($trace['file'], $trace['line'])) {
+                        $exceptions .= strip_tags($this->formatPath($trace['file'], $trace['line']));
+                    }
+                    $exceptions .= '</trace>';
+                }
+                $exceptions .= '</traces></exception>';
+            }
+            $exceptions .= '</exceptions>';
+        }
+
+        return <<<EOF
+<?xml version="1.0" encoding="{$this->charset}" ?>
+<problem xmlns="urn:ietf:rfc:7807">
+    <title>{$title}</title>
+    <status>{$statusCode}</status>
+    <detail>{$message}</detail>
+    {$exceptions}
+</problem>
+EOF;
+    }
+
+    /**
+     * Gets the TXT content associated with the given exception.
+     *
+     * @param \Exception|FlattenException $exception An \Exception or FlattenException instance
+     *
+     * @return string The TXT content as a string
+     */
+    public function getTxt($exception): string
+    {
+        if (!$exception instanceof FlattenException) {
+            $exception = FlattenException::create($exception);
+        }
+
+        if (404 === $statusCode = $exception->getStatusCode()) {
```

Comment:
this blocks seems repeated for all content-types? maybe move it to a private method?